base/rational.jl

```diff
@@ -127,7 +127,7 @@ function rationalize(::Type{T}, x::AbstractFloat, tol::Real) where T<:Integer
     if tol < 0
         throw(ArgumentError("negative tolerance $tol"))
     end
-    isnan(x) && return zero(T)//zero(T)
+    isnan(x) && return throw(ArgumentError("received NaN, not allowed for integer-related types"))
```

Comment:
Maybe just `"cannot rationalize NaN"`?

Reply:
There will almost certainly never be NaN for generic integer types...this is not how computer integers work.